### ISSUE TITLE
Manage Memberships API endpoint is not working after latest release upgrade

### ISSUE THREAD

**Issue by anarute** [loomio/loomio#9056]
After we upgraded to 2.15.0 the endpoint `/api/b1/memberships` to manage memberships is not working anymore.

I was able to reproduce locally and this is the error coming from the API:

<img width="620" alt="Screen Shot 2022-10-07 at 16 27 08" src="https://user-images.githubusercontent.com/333447/194577522-bb616239-3cd6-40bf-8e40-63212c2da49c.png">

In the server console:

```
Can't verify CSRF token authenticity.
method=POST path=/api/b1/memberships format=json controller=API::B1::MembershipsController action=create status=422 error='ActionController::InvalidAuthenticityToken: ActionController::InvalidAuthenticityToken' duration=2.78 view=0.00 db=0.00
  
ActionController::InvalidAuthenticityToken (ActionController::InvalidAuthenticityToken):
  
actionpack (6.1.4.6) lib/action_controller/metal/request_forgery_protection.rb:211:in `handle_unverified_request'
actionpack (6.1.4.6) lib/action_controller/metal/request_forgery_protection.rb:243:in `handle_unverified_request'
devise (4.8.1) lib/devise/controllers/helpers.rb:255:in `handle_unverified_request'
actionpack (6.1.4.6) lib/action_controller/metal/request_forgery_protection.rb:238:in `verify_authenticity_token'
activesupport (6.1.4.6) lib/active_support/callbacks.rb:427:in `block in make_lambda'
activesupport (6.1.4.6) lib/active_support/callbacks.rb:198:in `block (2 levels) in halting'
actionpack (6.1.4.6) lib/abstract_controller/callbacks.rb:34:in `block (2 levels) in <module:Callbacks>'
activesupport (6.1.4.6) lib/active_support/callbacks.rb:199:in `block in halting'
activesupport (6.1.4.6) lib/active_support/callbacks.rb:512:in `block in invoke_before'
activesupport (6.1.4.6) lib/active_support/callbacks.rb:512:in `each'
activesupport (6.1.4.6) lib/active_support/callbacks.rb:512:in `invoke_before'
activesupport (6.1.4.6) lib/active_support/callbacks.rb:115:in `block in run_callbacks'
activesupport (6.1.4.6) lib/active_support/callbacks.rb:137:in `run_callbacks'
actionpack (6.1.4.6) lib/abstract_controller/callbacks.rb:41:in `process_action'
actionpack (6.1.4.6) lib/action_controller/metal/rescue.rb:22:in `process_action'
actionpack (6.1.4.6) lib/action_controller/metal/instrumentation.rb:34:in `block in process_action'
activesupport (6.1.4.6) lib/active_support/notifications.rb:203:in `block in instrument'
activesupport (6.1.4.6) lib/active_support/notifications/instrumenter.rb:24:in `instrument'
activesupport (6.1.4.6) lib/active_support/notifications.rb:203:in `instrument'
actionpack (6.1.4.6) lib/action_controller/metal/instrumentation.rb:33:in `process_action'
actionpack (6.1.4.6) lib/action_controller/metal/params_wrapper.rb:249:in `process_action'
activerecord (6.1.4.6) lib/active_record/railties/controller_runtime.rb:27:in `process_action'
actionpack (6.1.4.6) lib/abstract_controller/base.rb:165:in `process'
actionview (6.1.4.6) lib/action_view/rendering.rb:39:in `process'
actionpack (6.1.4.6) lib/action_controller/metal.rb:190:in `dispatch'
actionpack (6.1.4.6) lib/action_controller/metal.rb:254:in `dispatch'
actionpack (6.1.4.6) lib/action_dispatch/routing/route_set.rb:50:in `dispatch'
actionpack (6.1.4.6) lib/action_dispatch/routing/route_set.rb:33:in `serve'
actionpack (6.1.4.6) lib/action_dispatch/journey/router.rb:50:in `block in serve'
actionpack (6.1.4.6) lib/action_dispatch/journey/router.rb:32:in `each'
actionpack (6.1.4.6) lib/action_dispatch/journey/router.rb:32:in `serve'
actionpack (6.1.4.6) lib/action_dispatch/routing/route_set.rb:842:in `call'
rack-attack (6.6.1) lib/rack/attack.rb:103:in `call'
http_accept_language (2.1.1) lib/http_accept_language/middleware.rb:14:in `call'
rack-attack (6.6.1) lib/rack/attack.rb:127:in `call'
rack (2.2.4) lib/rack/deflater.rb:44:in `call'
warden (1.2.9) lib/warden/manager.rb:36:in `block in call'
warden (1.2.9) lib/warden/manager.rb:34:in `catch'
warden (1.2.9) lib/warden/manager.rb:34:in `call'
rack (2.2.4) lib/rack/tempfile_reaper.rb:15:in `call'
rack (2.2.4) lib/rack/etag.rb:27:in `call'
rack (2.2.4) lib/rack/conditional_get.rb:40:in `call'
rack (2.2.4) lib/rack/head.rb:12:in `call'
actionpack (6.1.4.6) lib/action_dispatch/http/permissions_policy.rb:22:in `call'
actionpack (6.1.4.6) lib/action_dispatch/http/content_security_policy.rb:18:in `call'
rack (2.2.4) lib/rack/session/abstract/id.rb:266:in `context'
rack (2.2.4) lib/rack/session/abstract/id.rb:260:in `call'
actionpack (6.1.4.6) lib/action_dispatch/middleware/cookies.rb:689:in `call'
actionpack (6.1.4.6) lib/action_dispatch/middleware/callbacks.rb:27:in `block in call'
activesupport (6.1.4.6) lib/active_support/callbacks.rb:98:in `run_callbacks'
actionpack (6.1.4.6) lib/action_dispatch/middleware/callbacks.rb:26:in `call'
actionpack (6.1.4.6) lib/action_dispatch/middleware/executor.rb:14:in `call'
actionpack (6.1.4.6) lib/action_dispatch/middleware/actionable_exceptions.rb:18:in `call'
sentry-rails (5.4.2) lib/sentry/rails/rescued_exception_interceptor.rb:12:in `call'
actionpack (6.1.4.6) lib/action_dispatch/middleware/debug_exceptions.rb:29:in `call'
sentry-ruby (5.4.2) lib/sentry/rack/capture_exceptions.rb:28:in `block (2 levels) in call'
sentry-ruby (5.4.2) lib/sentry/hub.rb:199:in `with_session_tracking'
sentry-ruby (5.4.2) lib/sentry-ruby.rb:372:in `with_session_tracking'
sentry-ruby (5.4.2) lib/sentry/rack/capture_exceptions.rb:19:in `block in call'
sentry-ruby (5.4.2) lib/sentry/hub.rb:59:in `with_scope'
sentry-ruby (5.4.2) lib/sentry-ruby.rb:352:in `with_scope'
sentry-ruby (5.4.2) lib/sentry/rack/capture_exceptions.rb:18:in `call'
actionpack (6.1.4.6) lib/action_dispatch/middleware/show_exceptions.rb:33:in `call'
lograge (0.12.0) lib/lograge/rails_ext/rack/logger.rb:18:in `call_app'
railties (6.1.4.6) lib/rails/rack/logger.rb:26:in `block in call'
activesupport (6.1.4.6) lib/active_support/tagged_logging.rb:99:in `block in tagged'
activesupport (6.1.4.6) lib/active_support/tagged_logging.rb:37:in `tagged'
activesupport (6.1.4.6) lib/active_support/tagged_logging.rb:99:in `tagged'
railties (6.1.4.6) lib/rails/rack/logger.rb:26:in `call'
sprockets-rails (3.4.2) lib/sprockets/rails/quiet_assets.rb:13:in `call'
actionpack (6.1.4.6) lib/action_dispatch/middleware/remote_ip.rb:81:in `call'
request_store (1.5.1) lib/request_store/middleware.rb:19:in `call'
actionpack (6.1.4.6) lib/action_dispatch/middleware/request_id.rb:26:in `call'
rack (2.2.4) lib/rack/method_override.rb:24:in `call'
rack (2.2.4) lib/rack/runtime.rb:22:in `call'
activesupport (6.1.4.6) lib/active_support/cache/strategy/local_cache_middleware.rb:29:in `call'
actionpack (6.1.4.6) lib/action_dispatch/middleware/executor.rb:14:in `call'
actionpack (6.1.4.6) lib/action_dispatch/middleware/static.rb:24:in `call'
rack (2.2.4) lib/rack/sendfile.rb:110:in `call'
actionpack (6.1.4.6) lib/action_dispatch/middleware/host_authorization.rb:119:in `call'
railties (6.1.4.6) lib/rails/engine.rb:539:in `call'
puma (5.6.5) lib/puma/configuration.rb:252:in `call'
puma (5.6.5) lib/puma/request.rb:77:in `block in handle_request'
puma (5.6.5) lib/puma/thread_pool.rb:340:in `with_force_shutdown'
puma (5.6.5) lib/puma/request.rb:76:in `handle_request'
puma (5.6.5) lib/puma/server.rb:443:in `process_client'
puma (5.6.5) lib/puma/thread_pool.rb:147:in `block in spawn_thread'
```

The payload I sent in the request:

`{"emails"=>["ana@igalia.com"], "remove_absent"=>1, "api_key"=>"[FILTERED]"}`

Please note that the list memberships endpoint is working fine with the same API key, so the token is correct.

**Comment by anarute** [comment]
@robguthrie adding `config.action_controller.allow_forgery_protection = false` to the controllers config files solved the issue, but I'm not sure if that's a good idea. What do you think?

**Comment by robguthrie** [comment]
oh! sorry!  ill fix this this weekend.

**Comment by robguthrie** [comment]
Hi @anarute I've just published v2.15.1 which fixes this issue. 

Nice that you found a solution! However that makes Loomio vulnerable to forged requests. 

We can just skip authenticity token checking for the integration endpoints - which require an API key to use, so they're safe anyway. This is what I've done for 2.15.1

**Comment by anarute** [comment]
Thank you @robguthrie! I confirm that fixed the issue.
